plugin/sdk_v2/public/adapter/neko_adapter.py

```
@@ -0,0 +1,285 @@
+            return Err(InvalidArgumentError("failed to register adapter tool"))
+        if hasattr(self, "register_dynamic_entry"):
+            try:
+                registered = self.register_dynamic_entry(
+                    name,
+                    handler,
+                    name=display_name or name,
+                    description=description,
+                )
+                if inspect.isawaitable(registered):
+                    registered = await registered
+            except Exception as error:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)